subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -448,7 +449,14 @@ u8_t ll_adv_enable(u8_t enable)
 	lll = &adv->lll;
 
 	pdu_adv = lll_adv_data_peek(lll);
-	if (pdu_adv->tx_addr) {
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	rnd_type = (adv->own_addr_type&0x1);
```

Comment:
This is not required, see https://github.com/zephyrproject-rtos/zephyr/blob/635924ab66a401e3d369c42e2d5394e1b007f64f/subsys/bluetooth/controller/ll_sw/ull_adv.c#L176

unless, something broken in double buffering, that tx_addr is not set correctly.

Reply:
Well, the problem arises when the own addr type is set to RANDOM_PUBLIC_ID (0x2) in which case the tx_addr is updated by ull_filter_adv_pdu_update() and thus tx_addr == 1 but with rnd_addr all zeros. Thus the guard can fail, as I saw when executing the most recent (incl.  TSE 11172 - present in EBQ since 2019.1.7214) version of LL/SEC/ADV/BV-11C test case.

Reply:
Instead of me re-writing the test spec and trying to explain what the image in the TSE 11172 says, I suggest you look it up directly yourself. I'll email the document it directly to you @cvinayak 

I'll add here an image from the Ellisys Analyzer indicating the HCI commands:
![EBQ_SEC_ADV_BV-11](https://user-images.githubusercontent.com/40020610/70999804-6c4a7600-20da-11ea-87b8-946088294390.PNG)